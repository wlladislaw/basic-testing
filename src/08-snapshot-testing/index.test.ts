import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  test('should generate linked list from values 1', () => {
    const els = [77, 568, 1000];
    const resExpected = {
      value: 77,
      next: {
        value: 568,
        next: {
          value: 1000,
          next: {
            value: null,
            next: null,
          },
        },
      },
    };

    expect(generateLinkedList(els)).toStrictEqual(resExpected);
  });

  test('should generate linked list from values 2', () => {
    const els = [777, 422, 89, 99000];
    const res = generateLinkedList(els);

    expect(res).toMatchSnapshot();
  });
});
