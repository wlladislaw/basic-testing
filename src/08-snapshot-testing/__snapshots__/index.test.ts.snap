// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": {
          "next": null,
          "value": null,
        },
        "value": 99000,
      },
      "value": 89,
    },
    "value": 422,
  },
  "value": 777,
}
`;
